app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+              {
+                type: 'checkbox',
+                name: 'threatIds',
+                title:
+                  'What are the main threats that your commitment is seeking to address?',
+                description: 'Multi select option',
+                # defaultValue: [],
+                choices: [
+                  'Residential & commercial development',
+                  'Agriculture & aquaculture',
+                  'Energy production & mining',
+                  'Transportation & service corridors',
+                  'Biological resource use',
+                  'Human intrusions & disturbance',
+                  'Natural system modifications',
+                  'Invasive & other problematic species, genes & diseases',
+                  'Pollution',
+                  'Geological events',
+                  'Climate change & severe weather',
+                  'Other'
+                ]
+              },
+              {
+                type: 'paneldynamic',
+                name: 'linksAttributes',
+                title:
+                  'Please provide the name and URL of any other websites containing information on your commitment.',
+                description: '(Optional field)',
+                templateElements: [
+                  {
+                    type: 'text',
+                    name: 'linkName',
+                    titleLocation: 'hidden',
+                    placeHolder: 'Website name'
+                  },
+                  {
+                    type: 'text',
+                    name: 'linkUrl',
+                    titleLocation: 'hidden',
+                    placeHolder: 'https://www.example.com'
+                  }
+                ],
+                panelCount: 1,
+                confirmDelete: true,
+                confirmDeleteText: 'Are you sure you want to delete this website?',
+                panelAddText: 'Add another'
+              }
+            ],
+            title: 'Stage'
+          },
+          {
+            name: 'progressPage',
+            elements: [
+              {
+                type: 'paneldynamic',
+                name: 'progressItems',
+                title: 'Upload your progress documentation',
+                description: '(Optional field)',
+                templateElements: [
+                  {
+                    type: 'file',
+                    name: 'progressItemFilename',
+                    titleLocation: 'hidden',
+                    maxSize: 26_214_400
+                  },
+                  {
+                    type: 'comment',
+                    name: 'progressNotes',
+                    title: 'Option to include progress notes',
+                    description: '(Optional field)'
+                  }
+                ],
+                panelCount: 1,
+                confirmDelete: true,
+                confirmDeleteText: 'Are you sure you want to delete this file?',
+                panelAddText: 'Add another'
```

Comment:
see previous comment

Reply:
Think I updated this one a little including the names of fields and adding a placeholder for existing documents, as below, in case you want to update this version:

```
    {
      name: "progressPage",
      elements: [
        {
          type: "paneldynamic",
          name: "progressDocumentsAttributes",
          title: "Upload your progress documentation",
          description: "(Optional field)",
          templateElements: [
            {
              type: "file",
              name: "document",
              titleLocation: "hidden",
              maxSize: 26214400
            },
            {
              type: "comment",
              name: "progressNotes",
              title: "Option to include progress notes",
              description: "(Optional field)"
            }
          ],
          panelCount: 1,
          confirmDelete: true,
          confirmDeleteText: "Are you sure you want to delete this file?",
          panelAddText: "Add another"
        },
        {
          type: "html",
          name: "existingDocuments",
          html: '<p class="existingDocuments">Existing documents</p>'
        }
      ],
      title: "Progress"
    }
```